modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/SseEntryPoint.java

```diff
@@ -0,0 +1,186 @@
+    {
+        this.contextHolder = new SseContextHolder( sse );
+    }
+
+    @GET
+    @Path("events")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    public void subscribe( final @Context SseEventSink sseEventSink )
+    {
+        final SseContextHolder ctx = contextHolder;
+
+        if ( ctx == null )
+        {
+            return;
+        }
+
+        ctx.broadcaster.register( sseEventSink );
+
+        final OutboundSseEvent sseEvent = ctx.sse.newEventBuilder().
+            name( "list" ).
+            id( UUID.randomUUID().toString() ).
+            mediaType( MediaType.APPLICATION_JSON_TYPE ).
+            data( ListApplicationJson.class, new ListApplicationJson( applicationService.getInstalledApplications().stream().
+                map( application -> new ApplicationInfoJson( application, applicationService.isLocalApplication( application.getKey() ) ) ).
+                collect( Collectors.toList() ) ) ).build();
+
+        sseEventSink.send( sseEvent );
+    }
+
+    @Override
+    public void onEvent( final Event event )
+    {
+        if ( EVENT_TYPE.equals( event.getType() ) && event.isLocalOrigin() )
+        {
+            event.getValueAs( String.class, EVENT_TYPE_KEY ).
+                ifPresent( eventSubType -> {
+                    switch ( eventSubType )
+                    {
+                        case INSTALLED:
+                        case STATE:
+                        case UNINSTALLED:
+                            handleEvent( event, eventSubType );
+                            break;
+                        default:
+                            LOG.debug( "Ignoring {} {}", EVENT_TYPE, eventSubType );
+                            break;
+                    }
+                } );
+        }
+    }
+
+    private void handleEvent( final Event event, final String eventSubType )
+    {
+        final SseContextHolder ctx = contextHolder;
+
+        if ( ctx == null )
+        {
+            LOG.debug( "Skipping {} {}. Please to subscribe to send SSE events.", EVENT_TYPE, eventSubType );
+            return;
+        }
+
+        if ( event.getValue( APPLICATION_KEY_PARAM ).isPresent() )
+        {
+            final OutboundSseEvent.Builder eventBuilder = ctx.sse.newEventBuilder().
+                name( eventSubType ).
+                id( UUID.randomUUID().toString() ).
+                mediaType( MediaType.APPLICATION_JSON_TYPE );
+
+            if ( UNINSTALLED.equals( eventSubType ) )
+            {
+                eventBuilder.data( ApplicationUninstalledJson.class,
+                                   new ApplicationUninstalledJson( event.getValue( APPLICATION_KEY_PARAM ).get().toString() ) );
+            }
+            else
+            {
+                final ApplicationKey applicationKey = ApplicationKey.from( event.getValue( APPLICATION_KEY_PARAM ).get().toString() );
+
+                final Application application = applicationService.getInstalledApplication( applicationKey );
```

Comment:
It looks like overkill for state event (on start/stop) to fetch app details? But maybe @gbbirkisson has a use for it?

Reply:
I personally like the model K8s uses, that is send event type and the whole object related to that event. If it is not a huge performance problem, I think we should keep it. It will probably come in handy.

Reply:
In current implementation it is open for race conditions. That is my main concern.
But 
with installed event it is a race anyway
and race chance is low.